app/src/components/inputs/inputTagsSearch/inputTagsSearch.jsx

```diff
@@ -0,0 +1,144 @@
+    dynamicSearchPromptText: PropTypes.bool,
+  };
+  static defaultProps = {
+    uri: '',
+    options: [],
+    values: [],
+    focusPlaceholder: '',
+    loadingPlaceholder: '',
+    nothingFound: '',
+    isCreatable: false,
+    isAsync: false,
+    multi: true,
+    removeSelected: false,
+    makeOptions: () => {},
+    isValidNewOption: () => {},
+    validation: () => {},
+    onChange: () => {},
+    minLength: 1,
+    showNewLabel: false,
+    dynamicSearchPromptText: false,
+  };
+  state = {
+    searchPromptText: this.props.nothingFound,
+  };
+  onInputChange = (input) => {
+    const diff = this.props.minLength - input.length;
+    if (this.props.dynamicSearchPromptText && this.props.minLength && diff > 0) {
+      const dynamicSearchPromptText = (
+        <FormattedMessage
+          id={'InputTagsSearch.dynamicSearchPromptText'}
+          defaultMessage={'Please enter {length} or more characters'}
+          values={{ length: diff }}
+        />
+      );
+      this.setState({ searchPromptText: dynamicSearchPromptText });
+    } else {
+      this.setState({ searchPromptText: this.props.nothingFound });
+    }
+    return input;
+  };
+  getItems = (input) => {
+    if (input.length >= this.props.minLength) {
+      return fetch(`${this.props.uri}${input}`)
+        .then((response) => {
+          const options = this.props.makeOptions(response);
+          return ({ options });
+        });
+    }
+    return Promise.resolve({ options: [] });
+  };
+  isValidNewOption = ({ label }) => (
+    this.props.isValidNewOption(label, this.props.minLength, this.props.validation)
+  );
+  newItemLabel = (label) => {
+    if (this.props.showNewLabel) {
+      return (
+        <div>
+          <span>{label}</span>
+          <span className={cx('new')}><FormattedMessage id="InputTagsSearch.new" defaultMessage="New" /></span>
+        </div>
+      );
+    }
+    return label;
+  };
+  render() {
+    const { isAsync, isCreatable, loadingPlaceholder, focusPlaceholder,
+      values, options, onChange, multi, removeSelected } = this.props;
+    let SelectComponent;
+    if (isAsync) {
+      if (isCreatable) {
+        SelectComponent = AsyncCreatable;
+      } else {
+        SelectComponent = Async;
+      }
+    } else if (isCreatable) {
+      SelectComponent = Creatable;
+    } else {
+      SelectComponent = Select;
+    }
+    return (
+      <div className={cx('select-container')}>
+        <SelectComponent
+          loadOptions={this.getItems}
+          autoload={false}
+          cache={false}
+          className={cx('select2-search-tags')}
+          noResultsText={this.state.searchPromptText}
+          loadingPlaceholder={loadingPlaceholder}
+          searchPromptText={focusPlaceholder}
+          value={values}
+          options={options}
+          onInputChange={this.onInputChange}
+          onChange={onChange}
+          multi={multi}
+          optionRenderer={option => optionRenderer(option, onChange)}
```

Comment:
You can move `optionRenderer` into this class, and get onChange from props instead of passing it as an argument.
Also, rename it to `renderOption`.
So this line will become `optionRenderer={renderOption}`